circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -377,7 +470,18 @@ def _get_mapping_ids_by_partition(
         subcirc_map_ids.append([])
         for i, inst in enumerate(circ.data):
             if isinstance(inst.operation, SingleQubitQPDGate):
-                decomp_id = int(inst.operation.label.split("_")[-1])
+                try:
+                    decomp_id = int(inst.operation.label.split("_")[-1])
+                except (AttributeError, ValueError):
+                    raise ValueError(
+                        "SingleQubitQPDGate instances in input circuit(s) must have their "
+                        'labels suffixed with "_<id>", where <id> is the index of the gate '
+                        "relative to the other gates belonging to the same cut. For example, "
+                        "a two-qubit gate cut can be represented by two SingleQubitQPDGates -- one "
+                        'labeled "<your_label>_0" and one labeled "<your_label>_1".'
+                        "  This allows SingleQubitQPDGates belonging to the same cut to be "
+                        "sampled together."
```

Comment:
```suggestion
                        "sampled jointly."
```
i.e., the sampling is from the joint probability distribution (I think this change makes it more clear for people who recognize that term, at least.)